### ISSUE TITLE
Numbers and string literals not allowed as property names

### ISSUE THREAD

**Issue by brad4d** [google/closure-compiler#1851]
The spec says string literals and numeric literals should be allowed as property names in class definitions, but the compiler won't allow it.
https://tc39.github.io/ecma262/#prod-LiteralPropertyName

https://closure-compiler-debugger.appspot.com/#input0%3Dclass%2520Foo%2520%257B%250A%2520%2520'hello'()%2520%257B%2520return%2520'hello'%253B%2520%257D%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by MatrixFrog** [comment]
I seem to remember we did this on purpose to keep things simple. Anyone else have a similar memory? @matthewloring @blickly 

If it's not common in the wild I'd be happy to keep it this way (or maybe change the error to make it clear that we *know* it's technically valid and it's just our parser that doesn't accept it).

**Comment by matthewloring** [comment]
I don't believe I made a decision one way or the other. I don't recall explicitly disallowing something that was valid by spec but it's been a while.

![](https://media.giphy.com/media/FPjbHO0jJxGsE/giphy.gif)
